app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/CalendarAdapter.java

```diff
@@ -0,0 +1,125 @@
+package com.github.houseorganizer.houseorganizer.calendar;
+
+import static com.github.houseorganizer.houseorganizer.util.Util.logAndToast;
+
+import android.annotation.SuppressLint;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.github.houseorganizer.houseorganizer.R;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class CalendarAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>{
+    ActivityResultLauncher<String> getPicture;
+
+    Calendar calendar;
+    String eventToAttach;
+    final FirebaseStorage storage = FirebaseStorage.getInstance();
+    final FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    public CalendarAdapter(Calendar calendar, ActivityResultLauncher<String> getPicture) {
+        this.calendar = calendar;
+        this.getPicture = getPicture;
+        generateItems(calendar.getEvents());
+    }
+
+    public abstract CalendarAdapter switchView();
+
+    abstract void generateItems(List<Calendar.Event> events);
+
+    public Calendar getCalendar() {
+        return calendar;
+    }
+
+    public void showAddEventDialog(Context ctx, DocumentReference currentHouse, String errMessage) {
+        LayoutInflater inflater = LayoutInflater.from(ctx);
+        final View dialogView = inflater.inflate(R.layout.event_creation, null);
+        new AlertDialog.Builder(ctx)
+                .setTitle(R.string.event_creation_title)
+                .setView(dialogView)
+                .setPositiveButton(R.string.add, (dialog, id) -> {
+                    Task<DocumentReference> pushTask = pushEventFromDialog(dialogView, currentHouse);
+                    if (pushTask != null) {
+                            pushTask.addOnSuccessListener(documentReference -> refreshCalendarView(ctx, currentHouse, errMessage, calendar.getView() == Calendar.CalendarView.MONTHLY));
+                    }
+                    dialog.dismiss();
+                })
+                .setNegativeButton(R.string.cancel, (dialog, id) -> dialog.dismiss())
```

Comment:
I think it would also be nice if we do '.setCancelable(true);' so that if the user clicks in other parts of the screen the dialog disappears.